tests/ratio_estimator_test.py

```diff
@@ -72,3 +74,58 @@ def test_api_ratio_estimator(ratio_estimator, theta_shape, x_shape):
         nsamples,
     ), f"""unnormalized_log_ratio shape is not correct. It is of shape
     {unnormalized_log_ratio.shape}, but should be {(nsamples,)}"""
+
+
+def test_ratio_estimator_inference_with_correct_classifier():
```

Comment:
I suggest to combine these two separate tests into one that just as a `classifier_builder` argument. it uses a `pytest.mark.parametrize` to iterate over the three different cases you are covering here. additionally, for the two failing cases, you can use a `pytest.param`, e.g., sth like `build_classifier_missing_args(build_classifier_missing_args, marks=pytest.xfail(reason=..., raising...))` (please double check the syntax).

Do you see what I mean?